Tests/KituraWebSocketTests/WebSocketClient.swift

```diff
@@ -26,23 +27,37 @@ class WebSocketClient {
     let host: String
     let port: Int
     let uri: String
-    var channel: Channel? = nil
+    private var _channel: Channel? = nil
+    var delegate: WebSocketDelegate? = nil
+    let channelAccessQueue: DispatchQueue = DispatchQueue(label: "Channel Access Synchronization")
 
-    public init?(host: String, port: Int, uri: String, requestKey: String, onOpen: @escaping (Channel?) -> Void = { _ in }) {
+    public init?(host: String, port: Int, uri: String, requestKey: String) {
```

Comment:
We could have the initialiser accept a `WebSocketDelegate` object too, with the default value set to `DummyDelegate()`